llvm/test/Transforms/InstCombine/fmul-tan-cos.ll

```diff
@@ -0,0 +1,118 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
```

Comment:
Missing test for the commuted case `cos * tan` and a negative test with mismatched `X` (i.e. `tan(a) * cos(b)`).